### PR TITLE
Enable Fedora rawhide.

### PR DESCRIPTION
**Pull request by marxin**

### CODE REVIEW

**Comment by marxin** [comment]
@Conan-Kudo 

Any guess why we do see:
```
Run dnf --nogpgcheck --assumeyes install /usr/bin/appstream-util /usr/bin/cpio /usr/bin/bzip2 /usr/bin/desktop-file-validate /usr/bin/python3 /usr/bin/readelf /usr/bin/ldd /usr/bin/c++filt /usr/bin/xz dash devscripts-checkbashisms glibc glibc.i686 hunspell-en hunspell-cs python3-setuptools python3-enchant python3-magic python3-rpm python3-pybeam python3-pytest python3-pytest-cov python3-pytest-flake8 python3-pytest-xdist python3-flake8 python3-flake8-import-order python3-pyxdg python3-toml python3-zstd python3-pip rpm-build git
Fedora rawhide openh264 (From Cisco) - x86_64   0.0  B/s |   0  B     00:00    
Errors during downloading metadata for repository 'fedora-cisco-openh264':
  - Curl error (6): Couldn't resolve host name for https://mirrors.fedoraproject.org/metalink?repo=fedora-cisco-openh264-rawhide&arch=x86_64 [getaddrinfo() thread failed to start]
Error: Failed to download metadata for repo 'fedora-cisco-openh264': Cannot prepare internal mirrorlist: Curl error (6): Couldn't resolve host name for https://mirrors.fedoraproject.org/metalink?repo=fedora-cisco-openh264-rawhide&arch=x86_64 [getaddrinfo() thread failed to start]
Fedora - Rawhide - Developmental packages for t 0.0  B/s |   0  B     00:00    
Errors during downloading metadata for repository 'rawhide':
  - Curl error (6): Couldn't resolve host name for https://mirrors.fedoraproject.org/metalink?repo=rawhide&arch=x86_64 [getaddrinfo() thread failed to start]
  - Curl error (6): Couldn't resolve host name for https://mirrors.fedoraproject.org/metalink?repo=rawhide&arch=x86_64&countme=1 [getaddrinfo() thread failed to start]
Error: Failed to download metadata for repo 'rawhide': Cannot prepare internal mirrorlist: Curl error (6): Couldn't resolve host name for https://mirrors.fedoraproject.org/metalink?repo=rawhide&arch=x86_64 [getaddrinfo() thread failed to start]
```

?

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -14,8 +14,7 @@ jobs:
       matrix:
         docker:
           - 'registry.fedoraproject.org/fedora:latest'
-          # Temporarily disabled as Fedora is moving to Python 3.10 and it is broken now
-          # - 'registry.fedoraproject.org/fedora:rawhide'
+          - 'fedora:rawhide'
```
Comment (Conan-Kudo) [inline .github/workflows/main.yml:17]: Why this instead of `'registry.fedoraproject.org/fedora:rawhide'`?

**Review by Conan-Kudo — CHANGES_REQUESTED** [review]

**Comment by Conan-Kudo** [comment]
> @Conan-Kudo
> 
> Any guess why we do see:
> 
> ```
> Run dnf --nogpgcheck --assumeyes install /usr/bin/appstream-util /usr/bin/cpio /usr/bin/bzip2 /usr/bin/desktop-file-validate /usr/bin/python3 /usr/bin/readelf /usr/bin/ldd /usr/bin/c++filt /usr/bin/xz dash devscripts-checkbashisms glibc glibc.i686 hunspell-en hunspell-cs python3-setuptools python3-enchant python3-magic python3-rpm python3-pybeam python3-pytest python3-pytest-cov python3-pytest-flake8 python3-pytest-xdist python3-flake8 python3-flake8-import-order python3-pyxdg python3-toml python3-zstd python3-pip rpm-build git
> Fedora rawhide openh264 (From Cisco) - x86_64   0.0  B/s |   0  B     00:00    
> Errors during downloading metadata for repository 'fedora-cisco-openh264':
>   - Curl error (6): Couldn't resolve host name for https://mirrors.fedoraproject.org/metalink?repo=fedora-cisco-openh264-rawhide&arch=x86_64 [getaddrinfo() thread failed to start]
> Error: Failed to download metadata for repo 'fedora-cisco-openh264': Cannot prepare internal mirrorlist: Curl error (6): Couldn't resolve host name for https://mirrors.fedoraproject.org/metalink?repo=fedora-cisco-openh264-rawhide&arch=x86_64 [getaddrinfo() thread failed to start]
> Fedora - Rawhide - Developmental packages for t 0.0  B/s |   0  B     00:00    
> Errors during downloading metadata for repository 'rawhide':
>   - Curl error (6): Couldn't resolve host name for https://mirrors.fedoraproject.org/metalink?repo=rawhide&arch=x86_64 [getaddrinfo() thread failed to start]
>   - Curl error (6): Couldn't resolve host name for https://mirrors.fedoraproject.org/metalink?repo=rawhide&arch=x86_64&countme=1 [getaddrinfo() thread failed to start]
> Error: Failed to download metadata for repo 'rawhide': Cannot prepare internal mirrorlist: Curl error (6): Couldn't resolve host name for https://mirrors.fedoraproject.org/metalink?repo=rawhide&arch=x86_64 [getaddrinfo() thread failed to start]
> ```
> 
> ?

I wonder if that's related to the new glibc, maybe @fweimer would know?

File: .github/workflows/main.yml (hunk 1 above)
Comment (marxin) [reply .github/workflows/main.yml:17]: I tested that previously, but it leads to the mentioned error.

**Comment by fweimer** [comment]
This is an issue with Github Actions, filed as actions/virtual-environments#3812 (and closed without a fix so far).

**Comment by fweimer** [comment]
Perhaps something like aufover/aufover-benchmark#9 would apply here as well.

**Comment by marxin** [comment]
> Perhaps something like [aufover/aufover-benchmark#9](https://github.com/aufover/aufover-benchmark/pull/9) would apply here as well.

Thank you Florian, it helped.

**Comment by marxin** [comment]
Now it fails in `pytest`, due to:

```
______________________ ERROR collecting test/test_LSB.py _______________________
ImportError while importing test module '/__w/rpmlint/rpmlint/test/test_LSB.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.10/site-packages/_pytest/python.py:578: in _importtestmodule
    mod = import_path(self.fspath, mode=importmode)
/usr/lib/python3.10/site-packages/_pytest/pathlib.py:524: in import_path
    importlib.import_module(module_name)
/usr/lib64/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
/usr/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:170: in exec_module
    exec(co, module.__dict__)
test/test_LSB.py:2: in <module>
    from rpmlint.checks.LSBCheck import LSBCheck
E   ModuleNotFoundError: No module named 'rpmlint'
```

Apparently new Python search path has changed?

**Comment by Conan-Kudo** [comment]
@hroncok, do you know what's going on here, because I'm not sure what's happening...?

**Comment by hroncok** [comment]
I don't understand `cwd: /__w/rpmlint/rpmlint` -- shouldn't it be one above?

**Comment by hroncok** [comment]
Trying https://github.com/marxin/rpmlint/pull/2

**Comment by marxin** [comment]
> Trying [marxin#2](https://github.com/marxin/rpmlint/pull/2)

Please make a pull request against `rpm-software-management/rpmlint`.

**Comment by hroncok** [comment]
I am trying to make it on top of this.

**Comment by hroncok** [comment]
https://github.com/rpm-software-management/rpmlint/pull/682

**Pull request merged**